org.idempiere.test/src/org/idempiere/test/jasper/PrintWithinProcess.java

```diff
@@ -0,0 +1,144 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+ * GNU General Public License for more details.                        *
+ *                                                                     *
+ * You should have received a copy of the GNU General Public License   *
+ * along with this program; if not, write to the Free Software         *
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+ * MA 02110-1301, USA.                                                 *
+ **********************************************************************/
+package org.idempiere.test.jasper;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+import org.adempiere.exceptions.AdempiereException;
+import org.adempiere.util.ProcessUtil;
+import org.compiere.model.MInvoice;
+import org.compiere.model.MProcess;
+import org.compiere.model.PO;
+import org.compiere.model.Query;
+import org.compiere.process.ProcessInfo;
+import org.compiere.util.Env;
+import org.compiere.util.Trx;
+import org.idempiere.test.AbstractTestCase;
+import org.junit.jupiter.api.Test;
+
+/**
+ *
+ */
+public class PrintWithinProcess extends AbstractTestCase {
+
+	/**
+	 * 
+	 */
+	public PrintWithinProcess() {
+	}
+
+	@Test
+	public void testPrintWithinProcess() {
+		Properties ctx = Env.getCtx();
+		String trxName = getTrxName();
+
+		MProcess process = null;
+		try {
+			String fileName = putResourceInTempFolder("org/idempiere/test/jasper/AR_Invoice.jrxml");
+			process = new MProcess(ctx, 0, trxName);
+			process.set_ValueNoCheck("AD_Client_ID", 0);
+			process.setAD_Org_ID(0);
+			process.setJasperReport("file://" + fileName);
+			process.setName("Test Invoice Jasper");
+			process.setValue("Test_Invoice_Jasper");
+			process.saveCrossTenantSafeEx();
+			commit();
+
+			List<MInvoice> invoices = new Query(ctx, MInvoice.Table_Name, "C_Invoice_ID IN (?,?)", trxName)
+					.setClient_ID()
+					.setOnlyActiveRecords(true)
+					.setParameters(103, 109)
+					.list();
+			for (MInvoice invoice : invoices) {
+				invoice.setDescription("Test Printing within a Process");
+				invoice.saveEx();
+			}
+
+			ProcessInfo pi = new ProcessInfo (process.getName(), process.getAD_Process_ID());
+			pi.setClassName(ProcessUtil.JASPER_STARTER_CLASS);
+			pi.setAD_User_ID(getAD_User_ID());
+			pi.setAD_Client_ID(getAD_Client_ID());
+			pi.setPrintPreview(false);
+			pi.setIsBatch(true);
+			Trx trx = Trx.get(trxName, false);
+
+			List<File> pdfList = new ArrayList<File>();
+			for (MInvoice invoice : invoices) {
+				pi.setRecord_ID(invoice.getC_Invoice_ID());
+				ProcessUtil.startJavaProcess(Env.getCtx(), pi, trx, false);
+				assertFalse(pi.isError(), pi.getSummary());
+				pdfList.add(pi.getPDFReport());
```

Comment:
should assert pi.getPDFReport is not null

Reply:
ready